tests/virt_autotest/libvirt_routed_virtual_network.pm

```diff
@@ -70,17 +71,23 @@ sub run_test {
         #attach-interface via virsh worked for all sles guest
         $mac1   = '00:16:3e:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
         $model1 = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen')) ? 'netfront' : 'virtio';
-        assert_script_run("virsh attach-interface $guest network vnet_routed --model $model1 --mac $mac1 --live");
+        my $affecter = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen') && is_sle('=11-sp4')) ? '--persistent' : '';
+        assert_script_run("virsh attach-interface $guest network vnet_routed --model $model1 --mac $mac1 --live $affecter", 60);
 
         $mac2   = '00:16:3e:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
         $model2 = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen')) ? 'netfront' : 'virtio';
-        assert_script_run("virsh attach-interface $guest.clone network vnet_routed_clone --model $model2 --mac $mac2 --live");
-
-        test_network_interface("$guest",       mac => $mac1, gate => $gate1, target => $target1, net => "vnet_routed");
-        test_network_interface("$guest.clone", mac => $mac2, gate => $gate2, target => $target2, net => "vnet_routed_clone");
-
-        assert_script_run("virsh detach-interface $guest network --mac $mac1 --current");
-        assert_script_run("virsh detach-interface $guest.clone network --mac $mac2 --current");
+        assert_script_run("virsh attach-interface $guest.clone network vnet_routed_clone --model $model2 --mac $mac2 --live $affecter", 60);
+
+        #Wait for guests attached interface from virtual routed network
+        sleep 30;
+        my $net1 = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen') && is_sle('=11-sp4')) ? 'netfront' : 'vnet_routed';
+        test_network_interface("$guest", mac => $mac1, gate => $gate1, target => $target1, net => $net1);
+        my $net2 = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen') && is_sle('=11-sp4')) ? 'netfront' : 'vnet_routed_clone';
+        test_network_interface("$guest.clone", mac => $mac2, gate => $gate2, target => $target2, net => $net2);
+
+        my $exclusive = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen') && is_sle('=11-sp4')) ? 'bridge --live --persistent' : 'network --current';
```

Comment:
Why do you use `bridge` here? This should test two separated networks routed together.

Reply:
After attach virtual interface from our virtual network(e.g vnet_routed_clone), there was not virtual network name as SOURCE on sles11sp4 xen host, just only TYPE - bridge and mode - netfront from domiflist output. 
Such as:
virsh domiflist sles-15-sp2
IN  TYPE MODE MAC
--------------------------------------------
vif8 bridge netfront XX.XX.XX.XX

So, have to use with "bridge" to find out guest mac under save_guest_ip func on s11sp4 xen host here.